### PR TITLE
Add endpoint to return the network weight for POS chains

### PR DESCRIPTION
**Pull request by dangershony**
Endpoints to get the POS network difficulty to be used with the indexer for this issue
https://github.com/block-core/blockcore-indexer/issues/56

There are two approaches this can go:

- the approach of this PR where we have an additional endpoint for the POS difficulty and the current network weight.
- adding the network weight and difficulty to each block on the api method `GetBlock` (then on the explorer we can display the network weight on each block)

(note the difficulty on POS is calculated slightly differently when it comes to calculating the network weight)

### CODE REVIEW

File: src/Blockcore/Interfaces/INetworkWeight.cs [hunk 1]
```diff
@@ -0,0 +1,9 @@
+﻿using NBitcoin;
+
+namespace Blockcore.Interfaces
+{
+    public interface INetworkWeight
+    {
+        double GetPosNetworkWeight();
```
Comment (sondreb) [inline src/Blockcore/Interfaces/INetworkWeight.cs:7]: Do we call PoS for Pos or PoS in types?

File: src/Features/Blockcore.Features.Miner/Api/Controllers/StakingController.cs [hunk 2]
```diff
@@ -79,6 +79,31 @@ public IActionResult GetStakingInfo()
...
+        /// </summary>
+        /// <returns>All staking info details as per the GetStakingInfoModel.</returns>
+        [Route("getnetworkstakinginfo")]
+        [HttpGet]
+        public IActionResult GetNetworkStakingInfo()
+        {
+            try
+            {
+                if (!this.fullNode.Network.Consensus.IsProofOfStake)
+                    return ErrorHelpers.BuildErrorResponse(HttpStatusCode.MethodNotAllowed, "Method not allowed", "Method not available for Proof of Stake");
```
Comment (sondreb) [inline src/Features/Blockcore.Features.Miner/Api/Controllers/StakingController.cs:93]: The error message says PoS wrongly, it should be Proof-of-Work, right?

**Review by sondreb — CHANGES_REQUESTED** [review]
Good work! At minimum the error message must be fixed.

File: src/Blockcore/Interfaces/INetworkWeight.cs (hunk 1 above)
Comment (dangershony) [reply src/Blockcore/Interfaces/INetworkWeight.cs:7]: I think it follows the rest of the pattern like `PosConsensusFactory`

File: src/Features/Blockcore.Features.Miner/Api/Controllers/StakingController.cs (hunk 2 above)
Comment (dangershony) [reply src/Features/Blockcore.Features.Miner/Api/Controllers/StakingController.cs:93]: ah well spotted

**Comment by dangershony** [comment]
All tests passed now

**Pull request merged**